### PR TITLE
Encoder optimized settings

### PR DESCRIPTION
**Pull request by holiber**

### CODE REVIEW

**Comment by andycreeth** [comment]
something seems to have gotten messed up here with the diff

**Comment by holiber** [comment]
Waiting for backend team approval

**Comment by eddyStreamlabs** [comment]
Seems like one test failed and there are merge conflicts

**Comment by holiber** [comment]
Going push fixes today

File: app/services/settings/settings.ts [hunk 1]
```diff
@@ -241,7 +199,44 @@ export class SettingsService extends StatefulService<ISettingsState>
...
+    return option ? option.value : options[0].value;
+  }
+
+  private patchSetting(
+    settingsFormData: ISettingsSubCategory[],
+    name: string,
+    patch: Partial<IObsInput<TObsValue>>,
+  ) {
+    // tslint:disable-next-line
+    settingsFormData = cloneDeep(settingsFormData);
```
Comment (eddyStreamlabs) [inline app/services/settings/settings.ts]: confused by this line

Comment (holiber) [reply app/services/settings/settings.ts]: Cloning prevents this object to be changed outside this function

File: app/components/windows/EditStreamInfo.vue [hunk 2]
```diff
@@ -43,58 +43,38 @@
           :metadata="gameMetadata"
         />
       </h-form-group>
+      <h-form-group v-if="searchProfilesPending">
+        {{ $t('Checking optimized setting for') }} {{ gameModel.value }}...
```
Comment (gettinToasty) [inline app/components/windows/EditStreamInfo.vue]: this is stale, `gameModel` doesn't have a value anymore, it just is the value.

also, why is this wrapped in a form group?

File: app/components/windows/EditStreamInfo.vue.ts [hunk 3]
```diff
@@ -15,25 +15,23 @@ import { NavigationService } from 'services/navigation';
...
-  value: IEncoderPreset;
-  description: string;
-  longDescription: string;
-}
-
 @Component({
   components: {
     ModalLayout,
-    BoolInput,
+    FormInput,
```
Comment (gettinToasty) [inline app/components/windows/EditStreamInfo.vue.ts]: not sure why this is here

File: app/components/windows/EditStreamInfo.vue.ts [hunk 4]
```diff
@@ -15,25 +15,23 @@ import { NavigationService } from 'services/navigation';
...
     HFormGroup,
+    VFormGroup,
```
Comment (gettinToasty) [inline app/components/windows/EditStreamInfo.vue.ts]: or this

File: app/services/hosts.ts [hunk 5]
```diff
@@ -54,3 +55,15 @@ export class HostsService extends Service {
...
+  @Inject('HostsService') private hosts: HostsService;
+
+  get protocol() {
+    return Util.useLocalHost() ? 'http://' : 'https://';
+  }
+
+  getStreamlabsApi(endpoint: string) {
+    return `${this.protocol}${this.hosts.streamlabs}/api/v5/slobs/${endpoint}`;
+  }
+}
```
Comment (gettinToasty) [inline app/services/hosts.ts]: i'm not really sure why this exists

File: app/components/windows/EditStreamInfo.vue (hunk 2 above)
Comment (holiber) [reply app/components/windows/EditStreamInfo.vue]: Good catch. It's in `h-form-group` for the alignment under the input above

File: app/components/windows/EditStreamInfo.vue.ts (hunk 3 above)
Comment (holiber) [reply app/components/windows/EditStreamInfo.vue.ts]: Looks like something that I left here after the merge conflict resolving

File: app/services/hosts.ts (hunk 5 above)
Comment (holiber) [reply app/services/hosts.ts]: It's for getting API endpoints for the case when you have the backend on your local PC

Comment (holiber) [reply app/services/hosts.ts]: For example it's hard to use `https` when you have a localy installed backend

**Comment by holiber** [comment]
It's ok. Andy can't review this now and we're going to release this things on weekend

**Comment by holiber** [comment]
So he is going to add his comments after the merge

**Pull request merged**